### PR TITLE
Better display of Krylov solvers

### PR DESCRIPTION
**Pull request by amontoison**
@dpo @tmigot 
I modified the `show` function for the Krylov solvers.
It displays `nrows`, `ncols` AND the `storage` used by the solver!!!
Maybe we should say `unallocated` for the size of empty vectors instead of `0` byte.
What do you think? 

```julia
using SuiteSparseMatrixCollection, MatrixMarket, Krylov

ssmc = ssmc_db()

matrix_name = "pre2"
matrix = ssmc_matrices(ssmc, "", matrix_name)
paths = fetch_ssmc(matrix, format="MM")
M = MatrixMarket.mmread(joinpath(paths[1], matrix_name * ".mtx"))

bilq_solver = BilqSolver(size(M)..., Vector{Float64})
bicgstab_solver = BicgstabSolver(size(M)..., Vector{Float64})

matrix_name = "nlpkkt200"
matrix = ssmc_matrices(ssmc, "", matrix_name)
paths = fetch_ssmc(matrix, format="MM")
M = MatrixMarket.mmread(joinpath(paths[1], matrix_name * ".mtx"))

cg_solver = CgSolver(size(M)..., CuVector{Float64})
minresqlp_solver = MinresQlpSolver(size(M)..., CuVector{Float64})
```
Capture d’écran du 2022-10-07 02-07-58
Capture d’écran du 2022-10-07 02-08-14
Capture d’écran du 2022-10-07 02-12-08
Capture d’écran du 2022-10-07 02-12-24

### CODE REVIEW

**Comment by dpo** [comment]
Beautiful! How about something like this for the size: https://github.com/IainNZ/Humanize.jl/blob/master/src/Humanize.jl#L30 or https://discourse.julialang.org/t/how-to-print-human-readable-byte-counts/31877/2 ?

**Comment by amontoison** [comment]
`Base.format_bytes` is exactly what I needed!
Capture d’écran du 2022-10-07 11-30-47
Capture d’écran du 2022-10-07 11-30-56

**Comment by dpo** [comment]
Nice! Some tests are failing though.

**Comment by amontoison** [comment]
The tests are failing because I changed the `show` output and I didn't updated the tests.
I wanted some feedbacks before.

File: test/test_solvers.jl [hunk 1]
```diff
@@ -11,1175 +11,139 @@ function test_solvers(FC)
...
+          @test nsolution(solver) == 1
+        end
+
+        @test niter > 0
+        @test statistics(solver) === solver.stats
+        @test issolved(solver)
+      end
+
+      io = IOBuffer()
+      show(io, solver, show_stats=false)
```
Comment (tmigot) [inline test/test_solvers.jl:129]:
```suggestion
      show(io, solver, show_stats=true)
```
just to increase coverage :).

**Review by tmigot — APPROVED** [review]
Thanks!

File: test/test_solvers.jl (hunk 1 above)
Comment (amontoison) [reply test/test_solvers.jl:129]: You should look at line 147 ;)

File: src/krylov_solvers.jl [hunk 2]
```diff
@@ -1885,45 +1885,69 @@ for (KS, fun, nsol, nA, nAt, warm_start) in [
   end
 end
 
+function ksizeof(attribute)
+  if isa(attribute, Vector{<:AbstractVector}) && !isempty(attribute)
+    # A vector of vector is a vector of pointers in Julia.
```
Comment (amontoison) [inline src/krylov_solvers.jl:1890]:
```suggestion
    # A vector of vectors is a vector of pointers in Julia.
```

**Comment by amontoison** [comment]
Thanks for the feedbacks @tmigot and @dpo!

**Pull request merged**